From 1ce61c02eb59ba2a4e73613c947be207d7c4c731 Mon Sep 17 00:00:00 2001
From: Komal Bajaj <quic_kbajaj@quicinc.com>
Date: Fri, 17 Jan 2025 16:26:38 +0530
Subject: [PATCH] QCLINUX: arm64: dts: qcom: sa8775p: Add EL2 dtb support

Added EL2-specific DTBs for sa8775p and qcs9100 targets
provided through overlay.

Change-Id: Ic1f003d8166e63780255c9005c70ae1b1fdcf3e5
Signed-off-by: Komal Bajaj <quic_kbajaj@quicinc.com>
Co-developed-by: Mukesh Ojha <quic_mojha@quicinc.com>
Signed-off-by: Mukesh Ojha <quic_mojha@quicinc.com>
Upstream-Status: Inappropriate
---
 arch/arm64/boot/dts/qcom/Makefile         |  16 ++
 arch/arm64/boot/dts/qcom/sa8775p-el2.dtso | 272 ++++++++++++++++++++++
 2 files changed, 288 insertions(+)
 create mode 100644 arch/arm64/boot/dts/qcom/sa8775p-el2.dtso

diff --git a/arch/arm64/boot/dts/qcom/Makefile b/arch/arm64/boot/dts/qcom/Makefile
index 223f75692434..0c835c4e8ad0 100644
--- a/arch/arm64/boot/dts/qcom/Makefile
+++ b/arch/arm64/boot/dts/qcom/Makefile
@@ -97,7 +97,15 @@ dtb-$(CONFIG_ARCH_QCOM)	+= sa8155p-adp.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sa8295p-adp.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sa8540p-ride.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sa8775p-ride.dtb
+
+sa8775p-ride-el2-dtbs:= sa8775p-ride.dtb sa8775p-el2.dtbo
+
+dtb-$(CONFIG_ARCH_QCOM)	+= sa8775p-ride-el2.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sa8775p-ride-r3.dtb
+
+sa8775p-ride-r3-el2-dtbs:= sa8775p-ride-r3.dtb sa8775p-el2.dtbo
+
+dtb-$(CONFIG_ARCH_QCOM)	+= sa8775p-ride-r3-el2.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sc7180-acer-aspire1.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sc7180-idp.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sc7180-trogdor-coachz-r1.dtb
@@ -225,7 +233,15 @@ dtb-$(CONFIG_ARCH_QCOM)	+= sm8450-sony-xperia-nagara-pdx224.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sm8550-mtp.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sm8550-qrd.dtb
 dtb-$(CONFIG_ARCH_QCOM) += qcs9100-ride.dtb
+
+qcs9100-ride-el2-dtbs:= qcs9100-ride.dtb sa8775p-el2.dtbo
+
+dtb-$(CONFIG_ARCH_QCOM)	+= qcs9100-ride-el2.dtb
 dtb-$(CONFIG_ARCH_QCOM) += qcs9100-ride-r3.dtb
+
+qcs9100-ride-r3-el2-dtbs:= qcs9100-ride-r3.dtb sa8775p-el2.dtbo
+
+dtb-$(CONFIG_ARCH_QCOM)	+= qcs9100-ride-r3-el2.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= qcs9075-rb8.dtb
 dtb-$(CONFIG_ARCH_QCOM) += qcs9075-ride.dtb
 dtb-$(CONFIG_ARCH_QCOM) += qcs9075-ride-r3.dtb
diff --git a/arch/arm64/boot/dts/qcom/sa8775p-el2.dtso b/arch/arm64/boot/dts/qcom/sa8775p-el2.dtso
new file mode 100644
index 000000000000..b7278e5eeb20
--- /dev/null
+++ b/arch/arm64/boot/dts/qcom/sa8775p-el2.dtso
@@ -0,0 +1,272 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2025, Qualcomm Innovation Center, Inc. All rights reserved.
+ */
+
+/dts-v1/;
+/plugin/;
+
+&remoteproc_adsp {
+	iommus = <&apps_smmu 0x3000 0x0>;
+	qcom,devmem = <0x00110000 0x00110000 0x4000 0x1>,
+			<0x00123000 0x00123000 0x1000 0x3>,
+			<0x00124000 0x00124000 0x3000 0x3>,
+			<0x00127000 0x00127000 0x2000 0x3>,
+			<0x0012a000 0x0012a000 0x3000 0x3>,
+			<0x0012e000 0x0012e000 0x1000 0x3>,
+			<0x0012f000 0x0012f000 0x1000 0x1>,
+			<0x00144000 0x00144000 0x1000 0x1>,
+			<0x00148000 0x00148000 0x1000 0x1>,
+			<0x00149000 0x00149000 0xe000 0x3>,
+			<0x00157000 0x00157000 0x1000 0x3>,
+			<0x00158000 0x00158000 0xd000 0x3>,
+			<0x00165000 0x00165000 0x1000 0x3>,
+			<0x00172000 0x00172000 0x1000 0x3>,
+			<0x00173000 0x00173000 0x8000 0x3>,
+			<0x0017b000 0x0017b000 0x2000 0x3>,
+			<0x0017f000 0x0017f000 0x1000 0x3>,
+			<0x00184000 0x00184000 0x1000 0x1>,
+			<0x00186000 0x00186000 0x1000 0x1>,
+			<0x0019c000 0x0019c000 0x3000 0x3>,
+			<0x0019f000 0x0019f000 0xe000 0x3>,
+			<0x001ad000 0x001ad000 0x2000 0x3>,
+			<0x001b0000 0x001b0000 0x1000 0x3>,
+			<0x001b2000 0x001b2000 0x1000 0x3>,
+			<0x001b4000 0x001b4000 0x1000 0x3>,
+			<0x001b6000 0x001b6000 0x1000 0x3>,
+			<0x001be000 0x001be000 0x1000 0x3>,
+			<0x001bf000 0x001bf000 0x10000 0x3>,
+			<0x001cf000 0x001cf000 0x1000 0x3>,
+			<0x00403000 0x00403000 0x1000 0x3>,
+			<0x00443000 0x00443000 0x1000 0x3>,
+			<0x00483000 0x00483000 0x1000 0x3>,
+			<0x004c3000 0x004c3000 0x1000 0x3>,
+			<0x004fc000 0x004fc000 0x1000 0x1>,
+			<0x00780000 0x00780000 0x7000 0x1>,
+			<0x010d0000 0x010d0000 0x1000 0x1>,
+			<0x010d4000 0x010d4000 0x1000 0x1>,
+			<0x01dc4000 0x01dc4000 0x1000 0x1>,
+			<0x01dc5000 0x01dc5000 0x1000 0x3>,
+			<0x01dc8000 0x01dc8000 0x1000 0x3>,
+			<0x01dfa000 0x01dfa000 0x1000 0x3>,
+			<0x01dfd000 0x01dfd000 0x1000 0x3>,
+			<0x01f40000 0x01f40000 0x40000 0x3>,
+			<0x01fc0000 0x01fc0000 0x29000 0x1>,
+			<0x01fe9000 0x01fe9000 0x1000 0x3>,
+			<0x01fea000 0x01fea000 0x1000 0x1>,
+			<0x01feb000 0x01feb000 0x1000 0x1>,
+			<0x01fec000 0x01fec000 0x4000 0x1>,
+			<0x03c41000 0x03c41000 0x3000 0x3>,
+			<0x03c4f000 0x03c4f000 0x1000 0x3>,
+			<0x04001000 0x04001000 0x1000 0x3>,
+			<0x04002000 0x04002000 0x1000 0x3>,
+			<0x04041000 0x04041000 0x1000 0x3>,
+			<0x04042000 0x04042000 0x1000 0x3>,
+			<0x04043000 0x04043000 0x1000 0x3>,
+			<0x04045000 0x04045000 0x1000 0x3>,
+			<0x04046000 0x04046000 0x1000 0x3>,
+			<0x0404e000 0x0404e000 0x1000 0x3>,
+			<0x0404f000 0x0404f000 0x1000 0x3>,
+			<0x04b04000 0x04b04000 0x1000 0x3>,
+			<0x04b05000 0x04b05000 0x1000 0x3>,
+			<0x04b11000 0x04b11000 0x1000 0x3>,
+			<0x04b40000 0x04b40000 0x20000 0x3>,
+			<0x09080000 0x09080000 0x1000 0x3>,
+			<0x09200000 0x09200000 0x28000 0x3>,
+			<0x09300000 0x09300000 0x28000 0x3>,
+			<0x09400000 0x09400000 0x28000 0x3>,
+			<0x09500000 0x09500000 0x28000 0x3>,
+			<0x09600000 0x09600000 0x28000 0x3>,
+			<0x09700000 0x09700000 0x28000 0x3>,
+			<0x0b250000 0x0b250000 0x10000 0x3>,
+			<0x0b450000 0x0b450000 0x10000 0x3>,
+			<0x0bbf0000 0x0bbf0000 0x2000 0x1>,
+			<0x0c210000 0x0c210000 0x80000 0x1>,
+			<0x0c340000 0x0c340000 0x1000 0x3>,
+			<0x0c370000 0x0c370000 0x1000 0x1>,
+			<0x0c380000 0x0c380000 0x1000 0x1>,
+			<0x0c390000 0x0c390000 0x1000 0x1>,
+			<0x0c3a0000 0x0c3a0000 0x1000 0x1>,
+			<0x0c3b0000 0x0c3b0000 0x1000 0x1>,
+			<0x0c3c0000 0x0c3c0000 0x1000 0x1>,
+			<0x0c3d0000 0x0c3d0000 0x1000 0x1>,
+			<0x0c3e0000 0x0c3e0000 0x1000 0x3>,
+			<0x0c3f0000 0x0c3f0000 0x1000 0x1>,
+			<0x0c400000 0x0c400000 0x30000 0x3>,
+			<0x0c440000 0x0c440000 0x10000 0x3>,
+			<0x0c460000 0x0c460000 0x81000 0x1>,
+			<0x0e600000 0x0e600000 0x100000 0x3>,
+			<0x0e700000 0x0e700000 0xa0000 0x1>,
+			<0x0f000000 0x0f000000 0x1000000 0x3>,
+			<0x146d8000 0x146d8000 0x1000 0x3>,
+			<0x16000000 0x16000000 0x1000000 0x2>,
+			<0x80b00000 0x80b00000 0x100000 0x3>,
+			<0x90860000 0x90860000 0x20000 0x3>,
+			<0x90900000 0x90900000 0x200000 0x3>;
+};
+
+&remoteproc_cdsp0 {
+	iommus = <&apps_smmu 0x21c0 0x0400>;
+	qcom,devmem = <0x00110000   0x00110000   0x4000     0x1>,
+			<0x00114000   0x00114000   0xf000     0x3>,
+			<0x00123000   0x00123000   0x1000     0x3>,
+			<0x00124000   0x00124000   0x3000     0x3>,
+			<0x00127000   0x00127000   0x3000     0x3>,
+			<0x0012a000   0x0012a000   0x3000     0x3>,
+			<0x0012e000   0x0012e000   0x1000     0x3>,
+			<0x0012f000   0x0012f000   0x1000     0x1>,
+			<0x00144000   0x00144000   0x1000     0x1>,
+			<0x00148000   0x00148000   0x1000     0x1>,
+			<0x00149000   0x00149000   0xc000     0x3>,
+			<0x00155000   0x00155000   0x1000     0x3>,
+			<0x0016b000   0x0016b000   0x1000     0x3>,
+			<0x00172000   0x00172000   0x1000     0x3>,
+			<0x00173000   0x00173000   0x11000    0x3>,
+			<0x00184000   0x00184000   0x1000     0x1>,
+			<0x00186000   0x00186000   0x1000     0x1>,
+			<0x00187000   0x00187000   0x45000    0x3>,
+			<0x001cc000   0x001cc000   0x1000     0x3>,
+			<0x00406000   0x00406000   0x1000     0x3>,
+			<0x00412000   0x00412000   0x1000     0x3>,
+			<0x00446000   0x00446000   0x1000     0x3>,
+			<0x00452000   0x00452000   0x1000     0x3>,
+			<0x00486000   0x00486000   0x1000     0x3>,
+			<0x00492000   0x00492000   0x1000     0x3>,
+			<0x004c6000   0x004c6000   0x1000     0x3>,
+			<0x004d2000   0x004d2000   0x1000     0x3>,
+			<0x004fc000   0x004fc000   0x1000     0x1>,
+			<0x0063a000   0x0063a000   0x4000     0x3>,
+			<0x0063e000   0x0063e000   0x4000     0x3>,
+			<0x00780000   0x00780000   0x10000    0x1>,
+			<0x010d0000   0x010d0000   0x1000     0x1>,
+			<0x010da000   0x010da000   0x1000     0x1>,
+			<0x016c2000   0x016c2000   0x9000     0x3>,
+			<0x01703000   0x01703000   0x5000     0x3>,
+			<0x01f40000   0x01f40000   0x40000    0x3>,
+			<0x01fc0000   0x01fc0000   0x2b000    0x1>,
+			<0x01feb000   0x01feb000   0x1000     0x3>,
+			<0x01fec000   0x01fec000   0x4000     0x1>,
+			<0x01fff000   0x01fff000   0x1000     0x1>,
+			<0x09080000   0x09080000   0x1000     0x3>,
+			<0x090b4000   0x090b4000   0x1000     0x3>,
+			<0x090b8000   0x090b8000   0x1000     0x3>,
+			<0x09200000   0x09200000   0x28000    0x3>,
+			<0x09300000   0x09300000   0x28000    0x3>,
+			<0x09400000   0x09400000   0x28000    0x3>,
+			<0x09500000   0x09500000   0x28000    0x3>,
+			<0x09600000   0x09600000   0x28000    0x3>,
+			<0x09700000   0x09700000   0x28000    0x3>,
+			<0x0b2c0000   0x0b2c0000   0x10000    0x3>,
+			<0x0b2f0000   0x0b2f0000   0x10000    0x3>,
+			<0x0b4c0000   0x0b4c0000   0x10000    0x3>,
+			<0x0b4f0000   0x0b4f0000   0x10000    0x3>,
+			<0x0bbf0000   0x0bbf0000   0x2000     0x1>,
+			<0x0c20a000   0x0c20a000   0x1000     0x3>,
+			<0x0c20b000   0x0c20b000   0x1000     0x3>,
+			<0x0c210000   0x0c210000   0x80000    0x1>,
+			<0x0c330000   0x0c330000   0x1000     0x3>,
+			<0x0c370000   0x0c370000   0x1000     0x1>,
+			<0x0c380000   0x0c380000   0x1000     0x3>,
+			<0x0c390000   0x0c390000   0x1000     0x1>,
+			<0x0c3a0000   0x0c3a0000   0x1000     0x1>,
+			<0x0c3b0000   0x0c3b0000   0x1000     0x1>,
+			<0x0c3c0000   0x0c3c0000   0x1000     0x1>,
+			<0x0c3d0000   0x0c3d0000   0x1000     0x1>,
+			<0x0c3e0000   0x0c3e0000   0x1000     0x1>,
+			<0x0c3f0000   0x0c3f0000   0x1000     0x1>,
+			<0x0c440000   0x0c440000   0x10000    0x3>,
+			<0x0c460000   0x0c460000   0x81000    0x1>,
+			<0x0e600000   0x0e600000   0x100000   0x3>,
+			<0x0e700000   0x0e700000   0xa0000    0x1>,
+			<0x0f000000   0x0f000000   0x1000000  0x3>,
+			<0x146d8000   0x146d8000   0x1000     0x3>,
+			<0x16000000   0x16000000   0x1000000  0x2>,
+			<0x23007000   0x23007000   0x1000     0x1>,
+			<0x23047000   0x23047000   0x1000     0x1>,
+			<0x2304e000   0x2304e000   0x2000     0x3>,
+			<0x90860000   0x90860000   0x20000    0x3>,
+			<0x90900000   0x90900000   0x200000   0x3>;
+};
+
+&remoteproc_cdsp1 {
+	iommus = <&apps_smmu 0x29c0 0x0400>;
+	qcom,devmem = <0x00110000   0x00110000   0x4000     0x1>,
+			<0x00114000   0x00114000   0xf000     0x3>,
+			<0x00123000   0x00123000   0x1000     0x3>,
+			<0x00124000   0x00124000   0x3000     0x3>,
+			<0x00127000   0x00127000   0x3000     0x3>,
+			<0x0012a000   0x0012a000   0x3000     0x3>,
+			<0x0012e000   0x0012e000   0x1000     0x3>,
+			<0x0012f000   0x0012f000   0x1000     0x1>,
+			<0x00144000   0x00144000   0x1000     0x1>,
+			<0x00148000   0x00148000   0x1000     0x1>,
+			<0x00149000   0x00149000   0xc000     0x3>,
+			<0x00155000   0x00155000   0x1000     0x3>,
+			<0x0016b000   0x0016b000   0x1000     0x3>,
+			<0x00172000   0x00172000   0x1000     0x3>,
+			<0x00173000   0x00173000   0x11000    0x3>,
+			<0x00184000   0x00184000   0x1000     0x1>,
+			<0x00186000   0x00186000   0x1000     0x1>,
+			<0x00187000   0x00187000   0x45000    0x3>,
+			<0x001cc000   0x001cc000   0x1000     0x3>,
+			<0x00406000   0x00406000   0x1000     0x3>,
+			<0x00412000   0x00412000   0x1000     0x3>,
+			<0x00446000   0x00446000   0x1000     0x3>,
+			<0x00452000   0x00452000   0x1000     0x3>,
+			<0x00486000   0x00486000   0x1000     0x3>,
+			<0x00492000   0x00492000   0x1000     0x3>,
+			<0x004c6000   0x004c6000   0x1000     0x3>,
+			<0x004d2000   0x004d2000   0x1000     0x3>,
+			<0x004fc000   0x004fc000   0x1000     0x1>,
+			<0x0063a000   0x0063a000   0x4000     0x3>,
+			<0x0063e000   0x0063e000   0x4000     0x3>,
+			<0x00780000   0x00780000   0x10000    0x1>,
+			<0x010d0000   0x010d0000   0x1000     0x1>,
+			<0x010da000   0x010da000   0x1000     0x1>,
+			<0x016c2000   0x016c2000   0x9000     0x3>,
+			<0x01703000   0x01703000   0x5000     0x3>,
+			<0x01f40000   0x01f40000   0x40000    0x3>,
+			<0x01fc0000   0x01fc0000   0x2b000    0x1>,
+			<0x01feb000   0x01feb000   0x1000     0x3>,
+			<0x01fec000   0x01fec000   0x4000     0x1>,
+			<0x01fff000   0x01fff000   0x1000     0x1>,
+			<0x09080000   0x09080000   0x1000     0x3>,
+			<0x090b4000   0x090b4000   0x1000     0x3>,
+			<0x090b8000   0x090b8000   0x1000     0x3>,
+			<0x09200000   0x09200000   0x28000    0x3>,
+			<0x09300000   0x09300000   0x28000    0x3>,
+			<0x09400000   0x09400000   0x28000    0x3>,
+			<0x09500000   0x09500000   0x28000    0x3>,
+			<0x09600000   0x09600000   0x28000    0x3>,
+			<0x09700000   0x09700000   0x28000    0x3>,
+			<0x0b2c0000   0x0b2c0000   0x10000    0x3>,
+			<0x0b2f0000   0x0b2f0000   0x10000    0x3>,
+			<0x0b4c0000   0x0b4c0000   0x10000    0x3>,
+			<0x0b4f0000   0x0b4f0000   0x10000    0x3>,
+			<0x0bbf0000   0x0bbf0000   0x2000     0x1>,
+			<0x0c20a000   0x0c20a000   0x1000     0x3>,
+			<0x0c20b000   0x0c20b000   0x1000     0x3>,
+			<0x0c210000   0x0c210000   0x80000    0x1>,
+			<0x0c330000   0x0c330000   0x1000     0x3>,
+			<0x0c370000   0x0c370000   0x1000     0x1>,
+			<0x0c380000   0x0c380000   0x1000     0x3>,
+			<0x0c390000   0x0c390000   0x1000     0x1>,
+			<0x0c3a0000   0x0c3a0000   0x1000     0x1>,
+			<0x0c3b0000   0x0c3b0000   0x1000     0x1>,
+			<0x0c3c0000   0x0c3c0000   0x1000     0x1>,
+			<0x0c3d0000   0x0c3d0000   0x1000     0x1>,
+			<0x0c3e0000   0x0c3e0000   0x1000     0x1>,
+			<0x0c3f0000   0x0c3f0000   0x1000     0x1>,
+			<0x0c440000   0x0c440000   0x10000    0x3>,
+			<0x0c460000   0x0c460000   0x81000    0x1>,
+			<0x0e600000   0x0e600000   0x100000   0x3>,
+			<0x0e700000   0x0e700000   0xa0000    0x1>,
+			<0x0f000000   0x0f000000   0x1000000  0x3>,
+			<0x146d8000   0x146d8000   0x1000     0x3>,
+			<0x16000000   0x16000000   0x1000000  0x2>,
+			<0x23007000   0x23007000   0x1000     0x1>,
+			<0x23047000   0x23047000   0x1000     0x1>,
+			<0x2304e000   0x2304e000   0x2000     0x3>,
+			<0x90860000   0x90860000   0x20000    0x3>,
+			<0x90900000   0x90900000   0x200000   0x3>;
+};
-- 
2.46.0

